### PR TITLE
Drop support for EOL Python 2.6

### PR DESCRIPTION
**Pull request by hugovk**
Python 2.6 is EOL and no longer receiving security updates (or any updates) from the core Python team.

Version | Release date | Supported until
--- | ---------- | ----------
2.6 | 2008-10-01 | 2013-10-29

Source: https://en.wikipedia.org/wiki/CPython#Version_history

They're also little used.

Here's download stats from PyPI for uwsgi for last month, and earlier:

| category | percent | downloads |
|----------|--------:|----------:|
|      2.7 |  47.71% |   351,883 |
|      3.6 |  28.87% |   212,951 |
|      3.5 |  10.31% |    76,023 |
|      3.4 |   8.69% |    64,095 |
|      3.7 |   4.03% |    29,746 |
| null     |   0.29% |     2,108 |
|      3.3 |   0.06% |       459 |
|      2.6 |   0.04% |       287 |
|      3.8 |   0.00% |        29 |
|      3.2 |   0.00% |         2 |
| Total    |         |   737,583 |

image

https://pypistats.org/packages/uwsgi

### CODE REVIEW

File: plugins/python/uwsgi_python.h [hunk 1]
```diff
@@ -9,45 +9,19 @@
...
-#if PY_MAJOR_VERSION == 3 && PY_MINOR_VERSION == 0
-#define HAS_NO_ERRORS_IN_PyFile_FromFd
-#endif
-
 #if PY_MAJOR_VERSION == 3 && PY_MINOR_VERSION < 7
 #define HAS_NOT_PyOS_AfterFork_Child
 #endif
 
 #if PY_MAJOR_VERSION < 3
+#define HAS_NOT_PyFrame_GetLineNumber
```
Comment (nsoranzo) [inline plugins/python/uwsgi_python.h]: I don't think that's correct, this was not defined for Python 2.7 before.

Comment (hugovk) [reply plugins/python/uwsgi_python.h]: You're right, I've removed it. Thanks!

**Comment by hugovk** [comment]
One year on, stats for September 2019:

| category | percent | downloads |
|----------|--------:|----------:|
| 3.6      |  37.11% |   390,257 |
| 2.7      |  23.78% |   250,061 |
| 3.7      |  21.03% |   221,146 |
| 3.5      |   9.60% |   100,990 |
| 3.4      |   7.87% |    82,739 |
| null     |   0.55% |     5,803 |
| 3.8      |   0.04% |       473 |
| 2.6      |   0.01% |       112 |
| 3.9      |   0.00% |        12 |
| 3.3      |   0.00% |         8 |
| Total    |         | 1,051,601 |

Date range: 2019-09-01 - 2019-09-30

Source: `pip install -U pypistats && pypistats python_minor uwsgi --last-month`

**Comment by rdeioris** [comment]
Hi, thanks for you work, but take into account that uWSGI is a commercial product and we still have lot of customers using older versions of python (even 2.4!). So, for now we cannot remove it. I leave the pull request open as very probably we will drop the whole python2 support in 2.1/3.0.

Thanks again

**Comment by hugovk** [comment]
Thanks for the update! Do you have a rough idea of the schedule for 2.1/3.0? 

(I see the [uWSGI 2.1](https://github.com/unbit/uwsgi/milestone/4) milestone is "Past due by over 5 years" :)

**Comment by hugovk** [comment]
Closing to clear up my PRs list, let me know if you'd like to re-open it in the future. Thanks!

**Pull request closed without merging**